modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala

```diff
@@ -615,7 +633,16 @@ object ElasticRequest {
       val sortJson: Json =
         if (self.sortBy.nonEmpty) Obj("sort" -> Arr(self.sortBy.toList.map(_.paramsToJson): _*)) else Obj()
 
-      fromJson merge sizeJson merge highlightsJson merge sortJson merge self.query.toJson merge searchAfterJson
+      val sourceJson: Json =
+        included
+          .map(included => Obj("includes" -> Arr(included.map(_.toJson): _*)))
+          .map { included =>
+            included merge excluded.fold(Obj())(excluded => Obj("excludes" -> Arr(excluded.map(_.toJson): _*)))
+          }
+          .orElse(excluded.map(excluded => Obj("excludes" -> Arr(excluded.map(_.toJson): _*))))
+          .fold(Obj())(filters => Obj("_source" -> filters))
```

Comment:
I wonder if something like this could be a bit easier to read:
```
val sourceJson: Json = 
   (included, excluded) match {
      case (None, None) => Obj()
      case (in, ex) => 
         Obj( "_source" -> {
            val includes = included.fold(Obj())(included => Obj("includes" -> Arr(included.map(_.toJson): _*)))
            val excludes = excluded.fold(Obj())(excluded => Obj("excludes" -> Arr(excluded.map(_.toJson): _*))
            included merge excluded
         }
      }
```
Would this work?